### DISCUSSION TITLE
Spark error while predict

### DISCUSSION THREAD

**Discussion by robocon20x** [moj-analytical-services/splink#1475] [General]
Hi all, I run notebook about spark example, and I got this error
![image](https://github.com/moj-analytical-services/splink/assets/90462777/0d273416-81bd-4b8f-8637-ae32369a2ef5)
How can I fix it?
Second, it seem like that DuckDB is supported better than SparkDF, cause with the result is show cluster_id while not with Spark
![image](https://github.com/moj-analytical-services/splink/assets/90462777/075db9a3-7bb1-4b60-a5c2-93b02dd5ceab)
If i base on match_key value = 1 mean that they are the same, is it correct?
![image](https://github.com/moj-analytical-services/splink/assets/90462777/d13cc122-0c6e-4cee-aff8-28d368efaf0c)

**Comment by trduc79** [comment]
> Hi all, I run notebook about spark example, and I got this error

The better way is using python script and use vscode debugging feature to debug. The jupyter notebook in vscode limit the error field.
As a note that when you posting error, please COPY the error together with the image. The text is able to be indexed by search engine so other people with the same error can find it if it already solved.
And you should provide much more context instead of just one line of code that we can never think of what you are doing in the previous step. Create a small shorten, reproducible code and sample data is much helpful and IMHO is the only way other can troubleshoot for you. 

> Second, it seem like that DuckDB is supported better than SparkDF, cause with the result is show cluster_id while not with Spark

No, you may misunderstanding this column. This column is in the source data itself. The `cluster` is the column showing which records is predicted to belong to which group.

> If i base on match_key value = 1 mean that they are the same, is it correct?

Not really. It means they are matching by unique_id key column (correct me if I'm wrong).

**Reply by RobinL** [reply]
That's mostly right.

- You can cluster in both DuckDb and Spark using the `linker.cluster_pairwise_predictions_at_threshold` function
- match_key refers to the blocking rule that generated the comparison, it does not indicate they anything about whether the records are a match, you need `match_probability` or `match_weight` for that

As trduc79 said, to solve the error we'd need to see your full code and the full error message
